### PR TITLE
fix(compiler): report not existing files as errors

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/@angular/compiler-cli/src/codegen.ts [hunk 1]
```diff
@@ -131,7 +131,14 @@ export class CodeGenerator {
...
       reflectorHostContext?: ReflectorHostContext): CodeGenerator {
-    const xhr: compiler.XHR = {get: (s: string) => Promise.resolve(compilerHost.readFile(s))};
+    const xhr: compiler.XHR = {
+      get: (s: string) => {
+        if (!compilerHost.fileExists(s)) {
+          throw new Error(`File not found: ${s}`);
+        }
+        return Promise.resolve(compilerHost.readFile(s));
+      }
+    };
```
Comment (jelbourn) [inline modules/@angular/compiler-cli/src/codegen.ts]: No test?

Comment (tbosch) [inline modules/@angular/compiler-cli/src/codegen.ts]: We don't have a unit test setup for `codegen` yet. Especially for errors. That would require some more thinking that I can't do right now...

**Comment by jelbourn** [comment]
LGTM, I suppose, but maybe add a TODO for testing this behavior.

File: modules/@angular/compiler-cli/src/codegen.ts [hunk 2]
```diff
@@ -131,7 +131,14 @@ export class CodeGenerator {
   static create(
       options: AngularCompilerOptions, program: ts.Program, compilerHost: ts.CompilerHost,
       reflectorHostContext?: ReflectorHostContext): CodeGenerator {
-    const xhr: compiler.XHR = {get: (s: string) => Promise.resolve(compilerHost.readFile(s))};
+    const xhr: compiler.XHR = {
+      get: (s: string) => {
+        if (!compilerHost.fileExists(s)) {
+          throw new Error(`File not found: ${s}`);
```
Comment (IgorMinar) [inline modules/@angular/compiler-cli/src/codegen.ts]: Can you make the error even less ambiguous by prefixing it with "Compilation failed. Resource file not found: ..."

**Comment by IgorMinar** [comment]
lgtm otherwise

**Comment by tbosch** [comment]
Thanks. Changed the error message and added a TODO for adding tests.

File: modules/@angular/compiler-cli/src/codegen.ts [hunk 3]
```diff
@@ -131,7 +131,15 @@ export class CodeGenerator {
   static create(
       options: AngularCompilerOptions, program: ts.Program, compilerHost: ts.CompilerHost,
       reflectorHostContext?: ReflectorHostContext): CodeGenerator {
-    const xhr: compiler.XHR = {get: (s: string) => Promise.resolve(compilerHost.readFile(s))};
+    const xhr: compiler.XHR = {
+      get: (s: string) => {
+        if (!compilerHost.fileExists(s)) {
+          // TODO: We should really have a test for error cases like this!
+          throw new Error(`Compilation failed. Resource file not found: ${s}`);
```
Comment (alexeagle) [inline modules/@angular/compiler-cli/src/codegen.ts]: I think this is re-caught at https://github.com/angular/angular/blob/master/modules/%40angular/compiler-cli/src/main.ts#L31 with the "compilation failed" message so you maybe only need the "Resource file not found" in the message here? Check what the UI looks like for failures...

**Pull request merged**